### DISCUSSION TITLE
after update to cordova-android@11.0.0, app works on Android 12-13, but not Android 9-11

### DISCUSSION THREAD

**Discussion by vancew** [apache/cordova#381] [Q&A]
Hello,

I am getting the following error when I try to run my app on Android 9-11, even though it works on 12-13:
java.lang.RuntimeException: Unable to start activity ComponentInfo{appname/appname.MainActivity}: android.view.InflateException: Binary XML file line #24 in appname:layout/splash_screen_view: Failed to resolve attribute at index 0: TypedValue{t=0x2/d=0x7f0300ff a=-1}

I have used the advice from here:
https://stackoverflow.com/questions/73313620/ionic-android-12-splash-error-while-adding-cordova-android11/73399893
and here:
https://github.com/apache/cordova-android/issues/1497

and some great info from Norman Breau on the Slack channel which got me to the point I could build a working app for Android 12+.

I've tried all kinds of config changes trying to get past the above error and nothing seems to make it better, only worse.  At this point, I'd even gladly accept removing the splash screen on Android, just to get it to work on all versions again.

Please help.
AndroidManifest.xml:
```
<?xml version='1.0' encoding='utf-8'?>
<manifest android:hardwareAccelerated="true" android:versionCode="238" android:versionName="2.6.5" package="appname" xmlns:android="http://schemas.android.com/apk/res/android">
    <supports-screens android:anyDensity="true" android:largeScreens="true" android:normalScreens="true" android:resizeable="true" android:smallScreens="true" android:xlargeScreens="true" />
    <application android:allowBackup="false" android:exported="true" android:hardwareAccelerated="true" android:icon="@mipmap/ic_launcher" android:label="@string/app_name" android:networkSecurityConfig="@xml/network_security_config" android:requestLegacyExternalStorage="true" android:supportsRtl="true" android:testOnly="false">
        <activity android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|smallestScreenSize|screenLayout|uiMode" android:exported="true" android:label="@string/activity_name" android:launchMode="singleTop" android:name="MainActivity" android:theme="@style/Theme.App.SplashScreen" android:windowSoftInputMode="adjustResize">
            <intent-filter android:label="@string/launcher_name">
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
            <intent-filter>
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />
                <data android:scheme="appname" />
            </intent-filter>
        </activity>
        <service android:name="de.einfachhans.BackgroundMode.ForegroundService" />
        <provider android:authorities="${applicationId}.provider" android:exported="false" android:grantUriPermissions="true" android:name="org.apache.cordova.camera.FileProvider">
            <meta-data android:name="android.support.FILE_PROVIDER_PATHS" android:resource="@xml/camera_provider_paths" />
        </provider>
        <provider android:authorities="${applicationId}.emailcomposer.provider" android:exported="false" android:grantUriPermissions="true" android:name="de.appplant.cordova.emailcomposer.Provider">
            <meta-data android:name="android.support.FILE_PROVIDER_PATHS" android:resource="@xml/emailcomposer_provider_paths" />
        </provider>
        <provider android:authorities="${applicationId}.opener.provider" android:exported="false" android:grantUriPermissions="true" android:name="io.github.pwlin.cordova.plugins.fileopener2.FileProvider">
            <meta-data android:name="android.support.FILE_PROVIDER_PATHS" android:resource="@xml/opener_paths" />
        </provider>
        <provider android:authorities="${applicationId}.provider" android:exported="false" android:grantUriPermissions="true" android:name="de.appplant.cordova.plugin.notification.util.AssetProvider">
            <meta-data android:name="android.support.FILE_PROVIDER_PATHS" android:resource="@xml/localnotification_provider_paths" />
        </provider>
        <receiver android:exported="false" android:name="de.appplant.cordova.plugin.localnotification.TriggerReceiver" />
        <receiver android:exported="false" android:name="de.appplant.cordova.plugin.localnotification.ClearReceiver" />
        <activity android:exported="false" android:launchMode="singleInstance" android:name="de.appplant.cordova.plugin.localnotification.ClickReceiver" android:theme="@android:style/Theme.Translucent" />
        <activity android:configChanges="orientation|screenSize" android:label="Simple Audio Stream" android:name="com.hutchind.cordova.plugins.streamingmedia.SimpleAudioStream" />
        <activity android:configChanges="orientation|screenSize" android:label="Simple Video Stream" android:name="com.hutchind.cordova.plugins.streamingmedia.SimpleVideoStream" />
        <activity android:label="@string/multi_app_name" android:name="com.synconset.MultiImageChooserActivity" android:theme="@style/Theme.AppCompat.Light" />
        <receiver android:enabled="true" android:exported="true" android:name="nl.xservices.plugins.ShareChooserPendingIntent">
            <intent-filter>
                <action android:name="android.intent.action.SEND" />
            </intent-filter>
        </receiver>
        <provider android:authorities="${applicationId}.sharing.provider" android:exported="false" android:grantUriPermissions="true" android:name="nl.xservices.plugins.FileProvider">
            <meta-data android:name="android.support.FILE_PROVIDER_PATHS" android:resource="@xml/sharing_paths" />
        </provider>
    </application>
    <uses-permission android:name="android.permission.CAMERA" />
    <uses-feature android:name="android.hardware.camera" />
    <uses-feature android:name="android.hardware.camera.autofocus" />
    <uses-permission android:name="android.permission.WAKE_LOCK" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.ACCESS_MEDIA_LOCATION" />
    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
    <uses-feature android:name="android.hardware.location.gps" />
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
    <uses-permission android:name="android.permission.INTERNET" />
</manifest>

```

[config.xml.txt](https://github.com/apache/cordova/files/10689575/config.xml.txt)
[themes.xml.txt](https://github.com/apache/cordova/files/10689579/themes.xml.txt)
[colors.xml.txt](https://github.com/apache/cordova/files/10689580/colors.xml.txt)

**Comment by breautek** [comment]
> I'd even gladly accept removing the splash screen on Android, just to get it to work on all versions again.

Unfortunately this isn't really possible. You **could** go back to cordova-android@10 which doesn't handle android splashscreens. This has 3 caveats, without the splashscreen plugin you won't have a splashscreen for old android devices. For Android 12 (That's the version that introduced splashscreens if i recall) and later, you will get a splashscreen. If splashscreens are not defined (which it won't be in cordova-android@10), the android OS will fallback to use your app icon, which may or may not look good. This splashscreen is enforced, and in cordova-android@10 it's not controllable, short of adding/modifying the native project yourself.

The second caveat is without the splashscreen plugin, you won't have a JS api to control the splashscreen. If you include the splashscreen plugin, then on android 12+ you'll get double splashscreens (the plugin version and the OS native version).

The third caveat is, cordova-android@10 will have difficulties targeting API 31+ to satisify google play deployment requirements.

Onto your main issue, `appname:layout/splash_screen_view` I don't think is something added by Cordova. At least a find in project for `splash_screen_view` yields no results and I don't seem to have any reference in my own projects. Could it be a plugin adding it? Do you have any `layout` sources in your `<cordova-project>/platforms/android/app/src/main/res/layout` ??

**Reply by vancew** [reply]
Ah, thanks, yes, I didn't realize having a splash screen was basically required.  Upgrading to cordova-android@11 to get new versions deployed to the Google Play Store is what started these changes.

There were 4 files in the layout folder (that are part of cordova-plugin-telerik-imagepicker) :
actionbar_custom_view_done_discard.xml
actionbar_discard_button.xml
actionbar_done_button.xml
multiselectorgrid.xml

I removed that plugin and spent the day removing various plugins and then removing and re-adding cordova-android@11 and I'm still getting the error.  Thank you for your help so far!  Is there anything else you can think of?

**Reply by breautek** [reply]
Sorry for the late response, I just stumbled upon this again after naturally browsing through discussions.

On the bright side, I'm a bit more knowledgeable today compared to February.

If you haven't resolved your issue yet, I think the issue is with your `themes.xml`

You have:

```
<style name="Theme.App.SplashScreen" parent="@android:style/Theme.DeviceDefault.NoActionBar">
...
```

but the theme parent must be either `Theme.SplashScreen` or `Theme.SplashScreen.IconBackground`. Cordova defaults to `Theme.SplashScreen.IconBackground`, which allows you to have a background colour.

The full cordova example will look something like this:

```xml
<?xml version='1.0' encoding='utf-8'?>
<resources>
    <style name="Theme.App.SplashScreen" parent="Theme.SplashScreen.IconBackground">
        <item name="windowSplashScreenBackground">@color/cdv_splashscreen_background</item>
        <item name="windowSplashScreenAnimatedIcon">@drawable/ic_cdv_splashscreen</item>
        <item name="windowSplashScreenAnimationDuration">200</item>
        <item name="postSplashScreenTheme">@style/Theme.AppCompat.NoActionBar</item>
    </style>
</resources>
```

For more resources on the native side, see the [android docs](https://developer.android.com/develop/ui/views/launch/splash-screen/migrate).
